### PR TITLE
importccl: add strict_quotes option to CSV

### PR DESCRIPTION
**Pull request by spaskob**
importccl: add strict_quotes option to CSV

TL;DR this fixes a bwd compatibility breakage introduced in 19.2.
 
LazyQuotes is an option on the Go csv reader. We switched it
off for 19.2 but this breaks backward compatibility for clients
that relied on this behavior to parse lazily quotes in csv files.
With this PR we revert back to LazyQuotes being set to true by default
and we introduce an `WITH strict_quotes` option for `CSV IMPORT`
to allow to switch off LazyQuotes. Note that we'd like to keep
`strict_quotes` undocumented for now since the LazyQuotes mode is not very
well understood.

Release notes (sql change): add strict_quotes option to CSV.

### CODE REVIEW

**Comment by rolandcrosby** [comment]
Per conversation with @robert-s-lee, can we switch this to a `strict_quotes` option, off by default, that has the opposite effect?

**Comment by spaskob** [comment]
PTAL

**Review by rolandcrosby — APPROVED** [review]
> Reviewed 1 of 5 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/41610)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dt and @rolandcrosby)

**Comment by rolandcrosby** [comment]
LGTM, thanks for the quick fix!

**Review by dt — APPROVED** [review]
